tree/ntuple/v7/test/ntuple_join_table.cxx

```diff
@@ -246,3 +242,74 @@ TEST(RNTupleJoinTable, MultipleMatches)
    entryIdxs = joinTable->GetEntryIndexes({&(++run)});
    EXPECT_EQ(std::vector<ROOT::NTupleSize_t>{}, entryIdxs);
 }
+
+TEST(RNTupleJoinTable, Partitions)
+{
+   auto fnWriteNTuple = [](const FileRaii &fileGuard, std::uint32_t run) {
+      auto model = RNTupleModel::Create();
+      *model->MakeField<std::int16_t>("run") = run;
+      auto fldI = model->MakeField<std::uint32_t>("i");
+
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "ntuple", fileGuard.GetPath());
+
+      for (int i = 0; i < 5; ++i) {
+         *fldI = i;
+         ntuple->Fill();
+      }
+   };
+
+   auto joinTable = RNTupleJoinTable::Create({"i"});
+
+   std::vector<FileRaii> fileGuards;
+   fileGuards.emplace_back("test_ntuple_join_partition1.root");
+   fileGuards.emplace_back("test_ntuple_join_partition2.root");
+   fileGuards.emplace_back("test_ntuple_join_partition3.root");
+   fileGuards.emplace_back("test_ntuple_join_partition4.root");
+
+   std::int16_t runNumbers[4] = {1, 2, 3, 3};
+   std::vector<std::unique_ptr<RPageSource>> pageSources;
+
+   // Create three ntuples where with their corresponding run numbers and add them to the join table using this run
+   // number as the partition key.
+   for (unsigned i = 0; i < fileGuards.size(); ++i) {
+      fnWriteNTuple(fileGuards[i], runNumbers[i]);
+      pageSources.emplace_back(RPageSource::Create("ntuple", fileGuards[i].GetPath()));
+      joinTable->Add(*pageSources.back(), runNumbers[i]);
+   }
+
+   joinTable->Build();
+
+   std::vector<RNTupleOpenSpec> openSpec;
+   for (const auto &fileGuard : fileGuards) {
+      openSpec.emplace_back("ntuple", fileGuard.GetPath());
+   }
+   auto proc = RNTupleProcessor::CreateChain(openSpec);
+
+   auto i = proc->GetEntry().GetPtr<std::uint32_t>("i");
+   auto run = proc->GetEntry().GetPtr<int16_t>("run");
+
+   // Calling GetEntryIndexes with a partition key not present in the join table shouldn't fail; it should just return
+   // an empty vector.
+   EXPECT_EQ(std::vector<ROOT::NTupleSize_t>{}, joinTable->GetEntryIndexes({i.get()}, 4));
+
+   *i = 99;
+   // Similarly, calling GetEntryIndexes with a partition key that is present in the join table but a join value that
+   // isn't shouldn't fail; it should just return an empty vector.
+   EXPECT_EQ(std::vector<ROOT::NTupleSize_t>{}, joinTable->GetEntryIndexes({i.get()}, 4));
```

Comment:
But 4 isn't present, see the comment for the previous check. Did you mean to write
```suggestion
   *i = 99;
   // Similarly, calling GetEntryIndexes with a partition key that is present in the join table but a join value that
   // isn't shouldn't fail; it should just return an empty vector.
   EXPECT_EQ(std::vector<ROOT::NTupleSize_t>{}, joinTable->GetEntryIndexes({i.get()}, 3));
```
?